test/extensions/clusters/common/logical_host_integration_test.cc

```diff
@@ -0,0 +1,99 @@
+#include "source/common/config/utility.h"
+
+#include "test/integration/http_integration.h"
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+
+// Logical Host integration test.
+class LogicalHostIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                   public HttpIntegrationTest {
+public:
+  LogicalHostIntegrationTest()
+      : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()),
+        registered_dns_factory_(dns_resolver_factory_) {}
+
+  void createUpstreams() override { HttpIntegrationTest::createUpstreams(); }
+
+  NiceMock<Network::MockDnsResolverFactory> dns_resolver_factory_;
+  Registry::InjectFactory<Network::DnsResolverFactory> registered_dns_factory_;
+  uint32_t address_ = 0;
+  std::string address_string = "::1";
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, LogicalHostIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Reproduces a race from https://github.com/envoyproxy/envoy/issues/32850.
+// The test is by mocking the DNS resolver to return multiple different
+// addresses, also config dns_refresh_rate to be extremely fast.
+TEST_P(LogicalHostIntegrationTest, DISABLED_FastChangingDnsResult) {
+  // Only needs to test the race with one stack.
+  if (version_ == Network::Address::IpVersion::v4) {
```

Comment:
Do we need to use TEST_P then?

Reply:
Yes, added the support for both V4 and V6 test.